lib/handlers/health/apps.js

```diff
@@ -0,0 +1,130 @@
+var request = require('request');
+var async = require('async');
+var fhconfig = require('fh-config');
+var logger = require('../../util/logger').getLogger();
+
+var dfc;
+
+// Default timeout: 15 seconds
+var DEFAULT_TIMEOUT = 15*1000;
+// URI called to check if the app is reachable
+var DEFAULT_PING_URI = '/sys/info/ping';
+
+if (!fhconfig.value('openshift3')) {
+  var fhdfc = require('fh-dfc');
+
+  dfc = fhdfc(fhconfig.value('fhdfc'));
+}
+
+/**
+ * Utility function to build a result object
+ * @param status result status
+ * @param error optional error message
+ * @returns {{status: *, error: *}}
+ */
+function result(status, error) {
+  return {
+    status: status,
+    error: error
+  };
+}
+
+/**
+ *
+ * @param dynos The list of runnning dynos
+ * @param callback callback used to return the result of the operation. It must be in the format
+ * cb(err, appData) where appData will be a map with this structure:
+ * { appId: { dyno: dynoName, url: appUrl } }
+ */
+function getAppsUrl(dynos, callback) {
```

Comment:
I would extract some methods here, the indentation level is too deep. Something like:
```
async.each(dynos, doWhateverToDyno);
async.each(apps, doWhateverToApp);
```
So that anyone reading it for the first time doesn't have to spend time guessing.